boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -293,6 +293,26 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
             goto nextRange;
         }
 
+        /* Extra (input) safety. */
+        if (Regs.x.ecx != RTL_SIZEOF_THROUGH_FIELD(BIOS_MEMORY_MAP, Type) &&
+            Regs.x.ecx != sizeof(BIOS_MEMORY_MAP))
```

Comment:
I preferred the <, > checks. We're not in the business scolding bad BIOS writers, we just want the OS to work.

Reply:
The behavior change is for (unexpected) values in-between.
If we accept them, we get a (assumed) partly-uninitialized `Reserved`. How do we handle that then?
Or, do we simply assume we're safe because such a case will never happen?
Ftr, `Reserved` will later become ACPI 3.0 data.

Reply:
That argument applies exactly the same to the ecx == 20 case.
The proper way to call this interrupt is to pre-initialize Reserved to 1 of course, but that's not really related to this changeset.

Reply:
Agreed: 20/ACPI will be explicitly fixed/handled in a later PR. These two are valid BIOS output and missing ROS code.
But in-between values are not exactly the same case: they mean corrupted BIOS output.
Then, again, how do we (plan to) handle that, if not strictly rejecting the entry?
Automatically treat it as if it were a 20 reply? (And hope to be safe?)

Reply:
I really don't care, I'm done bikeshedding.

Reply:
Oh! Bikeshedding? I though we were discussing building Memory Map (in FreeLdr).
I already suggested 3 (future) options: explicit reject, implicit cast to 20, blindly cast to 24. Another (mixed) option could be to cast to 20 and unusable memory.
I am ready to code it the way you want, I am just looking for an explicit complete answer/choice.
Either you didn't give it or I didn't understand it.

Reply:
Eventually, I restored the check as a valid range, and added an explaining comment.